xs/src/libslic3r/Polyline.cpp

```diff
@@ -251,4 +252,97 @@ ThickPolyline::reverse()
     std::swap(this->endpoints.first, this->endpoints.second);
 }
 
+void
+concatThickPolylines(ThickPolylines& pp) {
+    bool changes = true;
+    while (changes){
+        changes = false;
+        //concat polyline if only 2 polyline at a point
+        for (size_t i = 0; i < pp.size(); ++i) {
+            ThickPolyline *polyline = &pp[i];
+
+            int32_t id_candidate_first_point = -1;
+            int32_t id_candidate_last_point = -1;
+            int32_t nbCandidate_first_point = 0;
+            int32_t nbCandidate_last_point = 0;
+            // find another polyline starting here
+            for (size_t j = 0; j < pp.size(); ++j) {
```

Comment:
this code should check for closed polylines and skip them (probably after setting endpoint to false)
Otherwise it will attach polylines to them (creating shape 6 or 8)

Reply:
if i can create a 6 with one line, i'm happy. Same for 8.

Reply:
Rest of code take takes care to avoid Y junctions, so I assumed there is reason for this.

There is another problem - 
creating 6 will choose some polyline as first and last candidate and following code will try to create loop.
creating 8 will find count 2 polylines at each end  (counting same polyline twice) and reject join